File: optimise/v04.c
#include <stdio.h>
#include <numeric>


void calculate(int N){
    int combinations = 0;
    int xx, yy;
    for(int x=2; x < N; x++){
        xx = x*x;
        for(int y=x+1; y < N; y++){
            yy = y*y;
            for(int z=y+1; z < N; z++){
                if(x < y && y < z &&
                   xx + yy == z * z &&
                   std::gcd(std::gcd(x, y), z) == 1){
                        combinations += 1;
                }
            }
        }
    }
    printf("%d\n", combinations);
}

int main(){
    int N;
    while (scanf("%d", &N) && N){
        calculate(N);
    }
    return 0;
}

File: optimise/v04.py
import sys
from math import gcd


for line in sys.stdin:
    N = int(line[:-1])
    combinations = 0
    for x in range(2, N + 1):
        for y in range(x + 1, N + 1):
            for z in range(y + 1, N + 1):
                if x * x + y * y == z * z and gcd(gcd(x, y), z) == 1:
                    combinations += 1
    print(combinations)
